spyder/utils/sourcecode.py

```diff
@@ -212,3 +221,188 @@ def shortest_path(files_path_list):
                 shortest_path_length = len(path_elmts)
                 shortest_path = path_elmts
         return os.path.join(*shortest_path)
+
+
+def shorten_paths(path_list, is_unsaved):

```

Comment:
Please move all new functions in this file to `spyder/utils/switcher.py`. They really don't belong here.